### PR TITLE
RegisterPage inputField, errormessage, noticemessage 수정하였습니다.

### PR DESCRIPTION
**Pull request by chepchep2**
인풋필드, 안내/에러 메세지 노출 수정입니다.
구현 결과입니다.
- https://imgur.com/Z8f8oI7

### CODE REVIEW

**Comment by chepchep2** [comment]
* **#64** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/64?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#65](https://github.com/team-stoodi/MyBaseballRecordClient/pull/65) <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/65?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#66](https://github.com/team-stoodi/MyBaseballRecordClient/pull/66) <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/66?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#63** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/63?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#62** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/62?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#61** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/61?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#60** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/60?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#59** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/59?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#58** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/58?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#57** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/57?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#56** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/56?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#55** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/55?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#54** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/54?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#53** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/53?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#52** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/52?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#51** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/51?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#50** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/50?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#49** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/49?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#48** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/48?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#47** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/47?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#46** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/46?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#45** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/45?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#44** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/44?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#43** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/43?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#42** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/42?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#41** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/41?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#40** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/40?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#39** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/39?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#35** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/35?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#34** <a href="https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/34?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/introPage`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @chepchep2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/page/register_page.dart [hunk 1]
```diff
@@ -48,14 +48,30 @@ class _RegisterPageState extends State<RegisterPage> {
...
+  }
+
   void _updateEmailStatus() {
     final email = _emailController.text;
     setState(() {
       _emailError =
           validateEmail(email) ? "" : AppTextList.criticalAlertEmailText;
-      _showPasswordInput = _emailError.isEmpty;
+      if (!_showPasswordInput && validateEmail(email)) {
+        _showPasswordInput = true;
```
Comment (flowkater) [inline lib/page/register_page.dart:71]:
```dart
_showPasswordInput = !_showPasswordInput && validateEmail(email)
```

File: lib/page/register_page.dart [hunk 2]
```diff
@@ -67,9 +83,11 @@ class _RegisterPageState extends State<RegisterPage> {
     setState(() {
       _passwordError =
           validatePassword(password) ? "" : AppTextList.passwordValidationText;
-      _showConfirmPasswordInput = _passwordError.isEmpty;
+      if (!_showConfirmPasswordInput && _passwordError.isEmpty) {
+        _showConfirmPasswordInput = true;
```
Comment (flowkater) [inline lib/page/register_page.dart:87]: 여기도 마찬가지

File: lib/page/register_page.dart [hunk 3]
```diff
@@ -84,7 +102,9 @@ class _RegisterPageState extends State<RegisterPage> {
           : "";
       _showPasswordCheckInput = _confirmPasswordError.isEmpty;
     });
-    if (_showPasswordCheckInput) {
+    if (_confirmPasswordError.isEmpty &&
```
Comment (flowkater) [inline lib/page/register_page.dart:105]: 이런 조건문 전부 private 함수로 빼기

**Review by flowkater — CHANGES_REQUESTED** [review]
여러개의 error 나 Input 이 존재할때 Map 으로 관리하면 좋을 것 같아요.

**Comment by flowkater** [comment]
키보드 올라올때 에러가 뜨네요.

File: lib/page/register_page.dart (hunk 1 above)
Comment (chepchep2) [reply lib/page/register_page.dart:71]: 조건을 충족해야만 _showPasswordInput = true가 되어야 하는데 
if (!_showPasswordInput && validateEmail(email)) {
        _showPasswordInput = true;
를 
_showPasswordInput = !_showPasswordInput && validateEmail(email)
로 바꾸면 해당 로직이 작동을 안하네요..

File: lib/page/register_page.dart (hunk 3 above)
Comment (chepchep2) [reply lib/page/register_page.dart:105]: 네 수정해보겠습니다.

**Comment by chepchep2** [comment]
이 부분에 대해서 조금 더 설명 부탁드려도 될까요?
Graphite에서는 안보이고 Github에서만 보여서 잘 이해가 안갔습니다.
저번 코드리뷰때 보여주셨던 _updateEmailStatus, _updatePasswordStatus 이런 메서드들을 한 개로 사용할때
말씀이실까요?

2024년 4월 3일 (수) 오후 11:13, flowkater ***@***.***>님이 작성:

File: lib/page/register_page.dart (hunk 1 above)
Comment (flowkater) [reply lib/page/register_page.dart:71]: 아 같은 변수라서 동작을 안할 것 같네요. 알겠습니다.

**Comment by chepchep2** [comment]
### Merge activity

* **Apr 7, 10:24 AM EDT**: @chepchep2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/59).
* **Apr 7, 11:47 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/59) rebased this pull request as part of a merge.
* **Apr 7, 11:48 AM EDT**: @chepchep2 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/team-stoodi/MyBaseballRecordClient/59).

**Pull request merged**